libc/test/include/header-test-template.c

```diff
@@ -0,0 +1,14 @@
+/*===-- Test for <@HEADER_NAME@> ----------------------------------------===
+ *
+ * Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See https://llvm.org/LICENSE.txt for license information.
+ * SPDSList-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <@HEADER_NAME@>
+
+#ifdef __cplusplus
+extern "C" int main() { return 0; }
+#else
+int main(void) { return 0; }
+#endif
```

Comment:
Do we need two copies of `main` like this?  Seeing the `extern "C"` on `main` is curious.

```suggestion
int main (int argc, char** argv) { return 0; }
```
Should be valid in both language modes.

Reply:
> Seeing the extern "C" on main is curious.

Yeah, just triple checked but `main` is treated specially; it's not name mangled in C++.
https://godbolt.org/z/5Ge551KEG